### PR TITLE
[17.0][MIG] geoengine_base_geolocalize: Migration to 17.0

### PR DESCRIPTION
**Pull request by peluko00**
Module migrated to version 17.0

cc https://github.com/APSL 155367
@miquelalzanillas @lbarry-apsl @javierobcn @mpascuall @BernatObrador @ppyczko  please review 

- [Issue] https://github.com/OCA/geospatial/issues/349

### CODE REVIEW

**Review by mpascuall — APPROVED** [review]
LGTM, tested in runboat

**Review by ppyczko — APPROVED** [review]
LGTM

**Comment by peluko00** [comment]
Could you review @yvaucher please. Thanks!

**Review by yvaucher — APPROVED** [review]

File: geoengine_base_geolocalize/tests/test_geoengine_partner.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class TestGeoenginePartner(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._super_send = requests.Session.send
+        super().setUpClass()
+
+    @classmethod
+    def _request_handler(cls, s, r, /, **kw):
+        """Don't block external requests."""
+        return cls._super_send(s, r, **kw)
```
Comment (yvaucher) [inline geoengine_base_geolocalize/tests/test_geoengine_partner.py:17]: This was a very bad idea. It makes the unit test calling an external service each time it is executed.

Comment (peluko00) [reply geoengine_base_geolocalize/tests/test_geoengine_partner.py:17]: Captura de pantalla 2026-02-26 a las 13 09 52
Hi @yvaucher!
According to OCA migration to 17.0.

Comment (yvaucher) [reply geoengine_base_geolocalize/tests/test_geoengine_partner.py:17]: Ouch, it's in OCA migration recommendations?

It's good practice to mock the query instead in unit test. Here is a good example, we got a ban error by OSM on some CI run as OSM was not happy being called numerous times.

To me the check made by Odoo completely makes sense and forces developpers to make better tests.
Depending on external resource is quite a bet on the stability of the tests.

Comment (yvaucher) [reply geoengine_base_geolocalize/tests/test_geoengine_partner.py:17]: I fixed the migration documentation to favor mocking in hope more dev will improve their tests. Thanks for the heads up.

**Pull request merged**